### DISCUSSION TITLE
Is it possible to get the exported mp3s to get the chapter titles?

### DISCUSSION THREAD

**Discussion by carl12** [audiamus/AaxAudioConverter#113] [Ideas]
This is one of the main inconveniences of using mp3s as opposed to directly in audible for me

**Comment by audiamus** [comment]
Well, the mp3 files do get chapter marks and titles. AAX Audio Converter creates them, thanks to ATL.net lib. This includes every output form, single file and any of the split modes. Chapter mark adjustments are also applied. However, you must find a player that supports MP3 chapters. Not many do, unfortunately. On Windows, for instance, VLC is MP3 chapter ignorant, but [MPC-HC](https://github.com/clsid2/mpc-hc) can read them.

If you can't find a player for your environment that honours MP3 chapters, have you tried to convert to MP4 instead, i.e. M4A or M4B, and find out whether your favourite player can handle that format?

**Reply by audiamus** [reply]
See also #104

**Comment by carl12** [comment]
I see the chapter number (like chapter 2) come across, but not the name of the chapter. In Audible, it is called Chapter 2: the Magicians.

![chapter2notitle](https://user-images.githubusercontent.com/4278181/151908166-e4b8f916-e555-44d1-b2b9-bd3cc0c14da0.png)

![chapter2title](https://user-images.githubusercontent.com/4278181/151908168-91d92f69-b0dc-4fa3-a2e4-a87df6df20f3.png)

**Reply by audiamus** [reply]
We have two sources for chapter meta data. 

One is embedded in the AAX file which is basically an MP4 container with an encrypted AAC audio stream. Hence it supports MP4 tags for meta data. Part of that meta data are the chapter tags. I call them embedded chapter data. While the tag specification would allow for individual chapter titles. Audible did not make use of this feature until very recently. They always numbered them in the generic form, Chapter 1, 2, 3 etc. Even for a multi-part book. Each part would start with 1 again.    

The other chapter source is an external meta data file. You get this extra data with the Audible App for Win10 which is now being discontinued. (Use the new [Book Lib Connect](https://github.com/audiamus/BookLibConnect) app as a substitute.) It's the `content_meta_<ASIN>.json` files that hold this extra chapter info. And these may have explicit chapter titles, but only if an Audible editor tended to that.

You can customize chapter processing in the the AAX Audio Converter settings, whether to use the JSON data or not. Because there may be catch: Chapter meta date also needs time stamps, called chapter marks, where a chapter actually starts. And in the external JSON files, quite often the chapter marks aren't accurate. AAX Audio Converter tries to fix this by silence detection and heuristic adjustment of chapter marks.  

- AAX file alone: Usually only simple chapter numbers, but almost always with accurate chapter marks. Only some of the most recent books have come with proper chapter titles.
- AAX file plus JSON meta data, via the Audible Win10 app or Book Lib Connect: A good chance for explicit chapter titles, but often at the cost of inaccurate chapter marks. Adjusting required. Some books still come with numbers only in the JSON file, even if the book source text has non-numerical chapters. 
 
Whatever chapter source and chapter processing is used, the result will always be transferred to MP3-transcoded output as well, as ID3v2 chapter tags.

**Reply by carl12** [reply]
> 

Thank you so much for your explanation. It's very helpful!
